cdap-watchdog/src/main/java/co/cask/cdap/logging/save/ExponentialBackoff.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.save;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Performs exponential backoff.
+ * Modified from Twill's SimpleKafkaConsumer class.
+ */
+public class ExponentialBackoff {
```

Comment:
I thought about it. We can do it later when there is another use for this class outside logging.